src/deck/drivers/src/aideck-router.c

```diff
@@ -62,25 +62,32 @@ static void cxpRxTest(void *param)
   while (1) {
     cpxReceivePacketBlocking(&cpxRx);
 
-    DEBUG_PRINT("CPX RX: Message from [0x%02X] to function [0x%02X] (size=%u)\n", cpxRx.route.source, cpxRx.route.function, cpxRx.dataLength);
-
-    if (cpxRx.route.source == CPX_T_ESP32) {
-      switch(cpxRx.route.function) {
-        case CPX_F_WIFI_CTRL:
-          if (cpxRx.data[0] == WIFI_AP_CONNECTED_CMD) {
-              DEBUG_PRINT("WiFi connected to ip: %u.%u.%u.%u\n",
-                          cpxRx.data[1],
-                          cpxRx.data[2],
-                          cpxRx.data[3],
-                          cpxRx.data[4]);
-          }
-          if (cpxRx.data[0] == WIFI_CLIENT_CONNECTED_CMD) {
-            DEBUG_PRINT("WiFi client connected\n");
-          }
-          break;
-        default:
-          DEBUG_PRINT("Not handling function [0x%02X]\n", cpxRx.route.function);
-      }
+    //DEBUG_PRINT("CPX RX: Message from [0x%02X] to function [0x%02X] (size=%u)\n", cpxRx.route.source, cpxRx.route.function, cpxRx.dataLength);
+
+    switch (cpxRx.route.function) {
+      case CPX_F_WIFI_CTRL:
+        if (cpxRx.data[0] == WIFI_AP_CONNECTED_CMD) {
+            DEBUG_PRINT("WiFi connected to ip: %u.%u.%u.%u\n",
+                        cpxRx.data[1],
+                        cpxRx.data[2],
+                        cpxRx.data[3],
+                        cpxRx.data[4]);
+        }
+        if (cpxRx.data[0] == WIFI_CLIENT_CONNECTED_CMD) {
+          DEBUG_PRINT("WiFi client connected\n");
+        }
+        break;
+      case CPX_F_CONSOLE:
+        if (cpxRx.route.source == CPX_T_ESP32) {
```

Comment:
Is it guaranteed that data is null-terminated?

Reply:
The ESP32 and GAP8 will send the null-terminator as well.